elide-core/src/main/java/com/yahoo/elide/core/filter/dialect/RSQLFilterDialect.java

```diff
@@ -0,0 +1,315 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * FilterDialect which implements support for RSQL filter dialect.
+ */
+public class RSQLFilterDialect implements SubqueryFilterDialect, JoinFilterDialect {
+
+    private final Pattern typedFilterPattern;
+    private final RSQLParser parser;
+    private final EntityDictionary dictionary;
+    private final Map<ComparisonOperator, Operator> operatorMap;
+
+    public RSQLFilterDialect(EntityDictionary dictionary, Set<ComparisonOperator> rsqlOperators) {
+        // Match "filter[<type>]"
+        typedFilterPattern = Pattern.compile("filter\\[([^\\]]+)\\]");
+
+        parser = new RSQLParser(rsqlOperators);
+        this. dictionary = dictionary;
+
+        /* Subuset of operators that map directly to Elide operators */
+        operatorMap = new HashMap<>();
+        operatorMap.put(RSQLOperators.IN, Operator.IN);
+        operatorMap.put(RSQLOperators.NOT_IN, Operator.NOT);
+        operatorMap.put(RSQLOperators.LESS_THAN, Operator.LT);
+        operatorMap.put(RSQLOperators.GREATER_THAN, Operator.GT);
+        operatorMap.put(RSQLOperators.GREATER_THAN_OR_EQUAL, Operator.GE);
+        operatorMap.put(RSQLOperators.LESS_THAN_OR_EQUAL, Operator.LE);
+    }
+
+    public RSQLFilterDialect(EntityDictionary dictionary) {
+        this(dictionary, RSQLOperators.defaultOperators());
+    }
+
+
+    @Override
+    public FilterExpression parseGlobalExpression(String path, MultivaluedMap<String, String> queryParams)
+            throws ParseException {
+        if (queryParams.size() != 1) {
+            throw new ParseException("There can only be a single filter query parameter");
+        }
+
+        MultivaluedMap.Entry<String, List<String>> entry = queryParams.entrySet().iterator().next();
+        String queryParamName = entry.getKey();
+
+        if (!queryParamName.equals("filter")) {
+            throw new ParseException("Invalid query parameter: " + queryParamName);
+        }
+        List<String> queryParamValues = entry.getValue();
+
+        if (queryParamValues.size() != 1) {
+            throw new ParseException("There can only be a single filter query parameter");
+        }
+
+        String queryParamValue = queryParamValues.get(0);
+
+        /* Extract the last collection in the URL */
+        path = Paths.get(path).normalize().toString().replace(File.separatorChar, '/');
+        if (path.startsWith("/")) {
+            path = path.substring(1);
+        }
+
+        String[] pathComponents = path.split("/");
+        String lastPathComponent = "";
+        if (pathComponents.length > 0) {
+            lastPathComponent = pathComponents[pathComponents.length - 1];
+        }
+
+        Class entityType = dictionary.getEntityClass(lastPathComponent);
```

Comment:
It works by virtue that global filters only work for root collections.  This could be fixed with a custom Visitor on the JSON-API path - that returns the 'type' of the last URL path segment.